kediatr-core/src/main/kotlin/com/trendyol/kediatr/AsyncQueryHandler.kt

```diff
@@ -1,5 +1,18 @@
 package com.trendyol.kediatr
 
-interface AsyncQueryHandler<R, Q : Query<R>> {
-    suspend fun handleAsync(query: Q): R
+/**
+ * Interface to be implemented for a non-blocking query handler
+ *
+ * @since 1.0.0
+ * @param TQuery any [Query] subclass to handle
+ * @see Query
+ * @see QueryHandler
+ */
+interface AsyncQueryHandler<TQuery : Query<TResponse>, TResponse> {
```

Comment:
Forgive me for late review. If I am not mistaken, @bilal-kilic this change in the order of type parameters introduces a breaking change for library users. 

Let say you're using Kediatr's current version and you want to update to this new version but it seems you need to rearrange the orders of parameters in the whole project :(